src/BloomExe/Publish/MakePdfUsingGeckofxHtmlToPdfProgram.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Windows.Forms;
+using Bloom.ToPalaso;
+using Palaso.CommandLineProcessing;
+using Palaso.IO;
+using Palaso.Progress;
+using System.Text;
+
+namespace Bloom.Publish
+{
+	/// <summary>
+	/// This wrapper uses the GeckoFxHtmlToPdf program.  Trying to use the component
+	/// directly leads to obscure bugs, at least on Windows.  Isolating the embedded
+	/// Gecko browser in a separate process appears to at least let us produce the
+	/// desired PDF files.
+	/// </summary>
+	/// <remarks>
+	/// The program always seems to fail after it finishes while xulrunner is trying
+	/// to clean up its COM objects.  Running the program standalone reveals this on
+	/// Linux by printing a scary SIGSEGV message and stack trace.  On Windows, it
+	/// brings up the "The Program Has Stopped Working" dialog which is even scarier.
+	/// (I have seen the corresponding Linux dialog once after running this program.)
+	/// However, the problem occurs after the desired output file has been safely
+	/// written to the disk, and is nicely hidden away from the user by running it
+	/// via CommandLineRunner.
+	/// </remarks>
+	class MakePdfUsingGeckofxHtmlToPdfProgram
+	{
+		public void MakePdf(string inputHtmlPath, string outputPdfPath, string paperSizeName,
+			bool landscape, Control owner, BackgroundWorker worker, DoWorkEventArgs doWorkEventArgs)
+		{
+			var runner = new CommandLineRunner();
+			string exePath;
+			var bldr = new StringBuilder();
+			var loc = Assembly.GetExecutingAssembly().Location;
+			var execDir = Path.GetDirectoryName(loc);
+			var fromDirectory = String.Empty;
+			if (Environment.OSVersion.Platform == PlatformID.Unix)
+			{
+				exePath = "/opt/mono-sil/bin/mono";
```

Comment:
Wouldn't it be sufficient to just use "mono"? I'd think that the PATH variable is set correctly from running Bloom so that it should use the correct version of mono. Advantage would be that it would use the same mono version that Bloom runs which would be helpful mainly for developers who might have mono in weird locations.